### DISCUSSION TITLE
Allow MQTT to be removed from build

### DISCUSSION THREAD

**Discussion by MaceMoneta** [arendst/Tasmota#14297] [Ideas]
I currently have about 85 Tasmota devices, but don't use MQTT. I use rsyslog action routines to trigger scripts based on log messages from the devices and wget to issue commands to them. It would be useful to be able to remove MQTT from the builds (not just disabled); this would significantly reduce the size of the binaries.

**Comment by sfromis** [comment]
Well, MQTT is literally in the name of Tasmota, the "M". And reduced size wins you nothing as long as it can fit. While basing an automation on syslog instead is possible, of course, I'd say that it is a rather rare use case. Anyway, as most of the code for MQTT is to create the JSON payloads you can also get from the log, the reduction would not be very significant.

**Reply by ascillato** [reply]
Hi, MQTT is the core of Tasmota. Tasmota is based on MQTT. Disabling is possible but a lot of routines are meant for MQTT, so allowing to not include MQTT would require a major rewrite of the core of this software. It might be possible, but not likely to happen anytime soon as it is the main feature. Besides, the MQTT libraries are very lightweight so the flash and RAM that you can free are not significant against the effort of removing them. It is recommended to disable unused features if you really need the space for another one.
